main.tf

```
@@ -63,7 +66,7 @@ resource "aws_iam_role" "default" {
 
 # https://docs.datadoghq.com/integrations/amazon_web_services/?tab=roledelegation#resource-collection
 resource "aws_iam_role_policy_attachment" "security_audit" {
-  count      = local.enabled && var.security_audit_policy_enabled ? 1 : 0
+  count      = local.enabled && (var.cspm_resource_collection_enabled || var.security_audit_policy_enabled) ? 1 : 0
```

Now that `var.cspm_resource_collection_enabled` is `null`, I don't think this will work in this logical expression, right ?

Another way to do this, by using the `datadog_integration_aws` resource and respecting the `enabled` logic, is to use a join splat expression. But again, I'm unsure what the default is of this value. I'm curious if `cspm_resource_collection_enabled` will return `false` or `true` if it's unset or `null` on the resource.

```suggestion
  count      = local.enabled && (join("", datadog_integration_aws.integration[*].cspm_resource_collection_enabled) || var.security_audit_policy_enabled) ? 1 : 0
```

---

I suppose it doesn't matter if `var.security_audit_policy_enabled` is `true`. I ran the tests to see if it works as expected.

---

Fixing now. Also seems like the Datadog API doesn't respond with any server-set default, at least according to the docs.